cardano-api/src/Cardano/Api/SpecialByron.hs

```diff
@@ -50,24 +70,27 @@ createByronUpdateProposal
   -> InstallerHash
   -> SigningKey ByronKey
   -> [ParametersToUpdate]
-  -> ByronUpdateProposal ()
+  -> ByronUpdateProposal
 createByronUpdateProposal nId pVer sVer sysTag insHash
                           (ByronSigningKey sKey) paramsToUpdate =
-    ByronUpdateProposal
-      $ signProposal (toByronProtocolMagicId nId) proposalBody noPassSigningKey
-  where
-    proposalBody :: ProposalBody
-    proposalBody = ProposalBody pVer protocolParamsUpdate sVer metaData
-
-    metaData :: M.Map SystemTag InstallerHash
-    metaData = M.singleton sysTag insHash
-
-    noPassSigningKey :: SafeSigner
-    noPassSigningKey = noPassSafeSigner sKey
-
-    protocolParamsUpdate :: ProtocolParametersUpdate
-    protocolParamsUpdate = createProtocolParametersUpdate
-                         emptyProtocolParametersUpdate paramsToUpdate
+  let nonAnnotatedProposal :: AProposal ()
+      nonAnnotatedProposal = signProposal (toByronProtocolMagicId nId) proposalBody noPassSigningKey
+      annotatedPropBody :: Binary.Annotated ProposalBody ByteString
+      annotatedPropBody = Binary.reAnnotate $ aBody nonAnnotatedProposal
```

Comment:
Blimey this is a bit ugly, but that's what the Byron ledger forces us into. It should also be safe because we enforce canonical encodings.

Perhaps this suggests that we should use `AProposal ()` (which is the same as the `Proposal` type alias) as our internal representation rather than `AProposal ByteString`. This stuff may come out cleaner that way.

Reply:
`ByronUpdateProposal` (from ouroboros-network) wants an `AProposal ByteString` so we would have to move this ugliness to `updateProposalToGenTx`

Reply:
We decided on slack to keep things as they are to avoid re-serialisation in `updateProposalToGenTx`

Reply:
:+1: